opentelemetry-sdk/src/testing/logs/in_memory_exporter.rs

```diff
@@ -0,0 +1,160 @@
+use crate::export::logs::{LogData, LogExporter};
+use async_trait::async_trait;
+use opentelemetry_api::logs::{LogError, LogResult};
+use std::sync::{Arc, Mutex};
+
+/// An in-memory logs exporter that stores logs data in memory..
+///
+/// This exporter is useful for testing and debugging purposes.
+/// It stores logs in a `Vec<LogData>`. Logs can be retrieved using
+/// `get_finished_logs` method.
+///
+/// # Example
+/// ```
+/// print(dupa)
+/// ```
+///
+#[derive(Clone, Debug)]
+pub struct InMemoryLogsExporter {
+    logs: Arc<Mutex<Vec<LogData>>>,
+}
+
+impl Default for InMemoryLogsExporter {
+    fn default() -> Self {
+        InMemoryLogsExporterBuilder::new().build()
+    }
+}
+
+///Builder for ['InMemoryLogsExporter'].
+/// # Example
+///
+/// ```
+/// use opentelemetry_sdk::testing::logs::{InMemoryLogsExporter, InMemoryLogsExporterBuilder};
+///
+/// use log::{error, info, Level, warn};
+/// use opentelemetry_appender_log::OpenTelemetryLogBridge;
+/// use opentelemetry_sdk::logs::{BatchLogProcessor, LoggerProvider};
+/// use opentelemetry_sdk::runtime;
+/// use opentelemetry_sdk::testing::logs::InMemoryLogsExporter;
+///
+///
+/// [tokio::main]
+/// sync fn main() {
+///    //Create an InMemoryLogsExporter
+///    let exporter: InMemoryLogsExporter = InMemoryLogsExporter::default();
+///    //Create a LoggerProvider and register the exporter
+///    let logger_provider = LoggerProvider::builder()
+///        .with_log_processor(BatchLogProcessor::builder(exporter.clone(), runtime::Tokio).build())
+///        .build();
+
+///    // Setup Log Appender for the log crate.
+///    let otel_log_appender = OpenTelemetryLogBridge::new(&logger_provider);
+///    log::set_boxed_logger(Box::new(otel_log_appender)).unwrap();
+///    log::set_max_level(Level::Info.to_level_filter());
+
+///    // Emit logs using macros from the log crate.
+///    error!("hello from {}. My price is {}", "apple", 2.99);
+///    warn!("warn!");
+///    info!("test log!");
+
+///    logger_provider.force_flush();
+
+///    let finished_logs = exporter.get_finished_logs().unwrap();
+///    for log in finished_logs {
+///        println!("{:?}", log);
+///    }
+///
+/// ```
+///
+#[derive(Debug, Clone)]
+pub struct InMemoryLogsExporterBuilder {}
+
+impl Default for InMemoryLogsExporterBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl InMemoryLogsExporterBuilder {
+    /// Creates a new instance of `InMemoryLogsExporter`.
+    ///
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    /// Creates a new instance of `InMemoryLogsExporter`.
+    ///
+    pub fn build(&self) -> InMemoryLogsExporter {
+        InMemoryLogsExporter {
+            logs: Arc::new(Mutex::new(Vec::new())),
+        }
+    }
+}
+
+impl InMemoryLogsExporter {
+    /// Returns the finished metrics as a vector of `LogData`.
```

Comment:
```suggestion
    /// Returns the logs emitted via Logger as a vector of `LogData`.
```

Reply:
Thx, didn't notice. I also thought that calling logs "finished" might be weird. So I changed the function and the naming everywhere to _emitted_.